src/main/java/com/microsoft/spring/data/gremlin/query/criteria/Criteria.java

```diff
@@ -24,21 +25,46 @@ private Criteria(CriteriaType type) {
         this.subCriteria = new ArrayList<>();
     }
 
-    public static Criteria getInstance(@NonNull String subject, CriteriaType type, @NonNull List<Object> values) {
+    private static boolean isBinaryOperation(CriteriaType type) {
+        switch (type) {
+            case AND:
+            case OR:
+                return true;
+            default:
+                return false;
+        }
+    }
+
+    private static boolean isUnaryOperation(CriteriaType type) {
+        switch (type) {
+            case IS_EQUAL:
+                return true;
+            default:
+                return false;
+        }
+    }
+
+    public static Criteria getUnaryInstance(CriteriaType type, @NonNull String subject, @NonNull List<Object> values) {
         final Criteria criteria = new Criteria(type);
 
+        Assert.isTrue(isUnaryOperation(type), "type should be Unary operation");
```

Comment:
How about do Assert check in the first line, before the `new` operation?

Reply:
thanks, should be at the beginning of method, miss that.